### PR TITLE
Handling fees with thresholds

### PR DESCRIPTION
**Pull request by dhriaznov**
# 🔘 PR Type
- [x] Enhancement

# 📜 Background
On mobile we have the lower fee and when the fee is higher limit, on the web-extension we don't, and adjusting the fee logic was left for later a couple of times.

Issue Link: #[[ENG-2680](https://linear.app/xverseapp/issue/ENG-2680/handling-fees-with-thresholds)]
Context Link (if applicable):

# 🔄 Changes
- Added high fees warning for transactions, comparing the fee with the high thresholds from the API.
- Changed the `feeMultipliers` property type in `WalletState` to the one that's actually fitting it. (Looked at our mobile app code as an example).

Impact:
- This PR impacts transactions (sending) for BTC (also Ordinals and BRC-20), STX (also Stacks NFTs and FT tokens)

# 🖼 Screenshot / 📹 Video
Monosnap Monosnap 2023-09-14 14-59-32

Monosnap Monosnap 2023-09-14 15-06-35

Monosnap Monosnap 2023-09-14 17-09-47

Monosnap Monosnap 2023-09-14 17-12-25

Monosnap Monosnap 2023-09-14 17-22-04

Monosnap Monosnap 2023-09-14 17-22-34

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/components/confirmBtcTransactionComponent/index.tsx [hunk 1]
```diff
@@ -280,7 +280,18 @@ function ConfirmBtcTransactionComponent({
...
     nonce?: string;
   }) => {
-    setCurrentFee(new BigNumber(modifiedFee));
+    const newFee = new BigNumber(modifiedFee);
+
+    if (
+      feeMultipliers &&
+      newFee.isGreaterThan(new BigNumber(feeMultipliers.thresholdHighSatsFee))
+    ) {
+      setShowFeeWarning(true);
```
Comment (abdulhaseeb4239) [inline src/app/components/confirmBtcTransactionComponent/index.tsx:289]: we also want to show the warning if the initial fee from transaction is greater than the threshold.

File: src/locales/en.json [hunk 2]
```diff
@@ -296,14 +296,14 @@
...
     "SCRIPT_OUTPUT_TX": "This transaction contains a script output. Be sure you trust the source of this transaction before confirming it.",
-
     "ORDINAL": "Ordinal",
     "ORDINAL_DETECTED_WARNING": "Ordinal inscription detected in transaction. Continuing will cause your ordinal to be transferred away.",
     "ORDINAL_DETECTED_ACTION": "Move to my ordinals address",
     "BTC_TRANSFER_DANGER_ALERT_TITLE": "Danger",
     "BTC_TRANSFER_DANGER_ALERT_DESC": "You are about to make a Bitcoin transfer which contains an ordinal inscription. Once transferred out of the wallet, you will not be able to recover them.",
     "CONITNUE": "Continue",
     "BACK": "Back",
+    "HIGH_FEE_WARNING_TEXT": "The estimated transaction fees for this transaction are very high.",
```
Comment (abdulhaseeb4239) [inline src/locales/en.json:306]: do we want to add this in other language files?

**Review by abdulhaseeb4239 — CHANGES_REQUESTED** [review]

File: src/locales/en.json (hunk 2 above)
Comment (dhriaznov) [reply src/locales/en.json:306]: Currently, we only have this file with locale keys in web extension repo 🙂

File: src/app/components/confirmBtcTransactionComponent/index.tsx (hunk 1 above)
Comment (dhriaznov) [reply src/app/components/confirmBtcTransactionComponent/index.tsx]: I updated the code so now it should show it for the initial fee as well

**Review by abdulhaseeb4239 — APPROVED** [review]

**Comment by blehitsahmad** [comment]
@dhriaznov left comments in the linear ticket

**Comment by dhriaznov** [comment]
I updated the PR adding the high warning fee for 1-step brc-20 transactions, could you approve this once again, please? @abdulhaseeb4239

**Comment by blehitsahmad** [comment]
@dhriaznov good to merge, added the comment to the [linear ticket](https://linear.app/xverseapp/issue/ENG-2680/handling-fees-with-thresholds)

**Review by Imamah-Zafar — APPROVED** [review]

**Review by yknl — APPROVED** [review]

**Pull request merged**